### DISCUSSION TITLE
Man page command-line switch case change

### DISCUSSION THREAD

**Discussion by Elliria** [autokey/autokey#840] [General]
The single-character command-line switch that's used to display [the AutoKey version](https://github.com/autokey/autokey/blob/develop/lib/autokey/argument_parser.py#L52-L56) in the **autokey-gtk** and **autokey-qt** man pages requires an **upper-case** **V** instead of a lower-case **v**. That's different from the other command-line switches (as can be seen in the code above and below the selection link to above) and seems to be counter-intuitive (at least for me). It would also be an easy fix.

Are there any negative consequences that would come from making that change or should I go ahead and create an issue for it and fix the issue?

**Comment by josephj11** [comment]
This is a bit weird. autokey-qt  works fine with --version and not with --Version, but does require -V as a short option.

May have something to do with --verbose also starting with a v.
Most programs use -v for verbose,  but AutoKey doesn't offer that as a short option, using -l instead.

We need to rethink how this should work.
-l depreciated (but still working for a while)
-v verbose
-V version

is a first approximation for straightening this out.

```
$ autokey-qt --version
autokey-qt Version 0.96.0-beta.10
bigbird@sananda:~/pq
$ autokey-qt --Version
usage: autokey-qt [-h] [-l] [-c] [-V] [--cutelog-integration] [-m]
autokey-qt: error: unrecognized arguments: --Version
bigbird@sananda:~/pq
$ autokey-qt -v
usage: autokey-qt [-h] [-l] [-c] [-V] [--cutelog-integration] [-m]
autokey-qt: error: unrecognized arguments: -v
bigbird@sananda:~/pq
$ autokey-qt -V
autokey-qt Version 0.96.0-beta.10
bigbird@sananda:~/pq
```

--cutelog-integration is another one I've never seen before. Any idea what that's about?

**Reply by Elliria** [reply]
> This is a bit weird. autokey-qt works fine with --version and not with --Version, but does require -V as a short option.

Yes. If you look at its [source code](https://github.com/autokey/autokey/blob/master/lib/autokey/argument_parser.py#L51-L56), you can see that it's asking for **"-V", "--version"**.
 
> May have something to do with --verbose also starting with a v. Most programs use -v for verbose, but AutoKey doesn't offer that as a short option, using -l instead.

Yeah. That's odd.
 
> We need to rethink how this should work. -l depreciated (but still working for a while) -v verbose -V version
>
> is a first approximation for straightening this out.

I'm not liking that one capital switch when all the other switches are lower-case. Maybe we could change version to release and then the options would be **-r** and **--release** and the description could be "Print the current AutoKey release/version to standard output and then exit."
 
> --cutelog-integration is another one I've never seen before. Any idea what that's about?

Yep. There's a link to [https://github.com/busimus/cutelog](https://github.com/busimus/cutelog) in the new man page for more details. It's a graphical log viewer.

**Comment by josephj11** [comment]
`--version` is common throughout Linux, so I'd rather not change that. "release' could be a noun or a verb, so it's also less than ideal because of that as well.

As for the capital V, I'm open to suggestions. Lots of commands have mixed case options where the lower and upper case letters are often totally different options, so it's no big deal to me.

I'll have to check out the log viewer.

**Comment by Elliria** [comment]
Then getting rid of -l for verbose and doing the two **v** entries this way, as you suggested, would be the best solution:
* -v, --verbose
* -V, --version

At least the capital V would then make sense.

**Reply by josephj11** [reply]
Works for me with the provision that -l should be depreciated, not eliminated - which means it still works and shows up in documentation and in help - possibly with a message that it's depereciated and will be removed in a future release when it is actually used.

-v, --verbose, (-l depreciated)

looks weird, but I don't see a better way.

**Reply by Elliria** [reply]
Good idea. Hopefully, @sebastiansam55 or @BlueDrink9 will chime in here to let us know if there would be any repercussions to changing the command-line switches. Also, I'll need to do a quick bit of research to find out whether more than two switches can be listed (**-v, --verbose, -l** ). This should be an easy fix, though.

**Reply by sebastiansam55** [reply]
I mean the only repercussion would be the obvious that they no longer work. 

I think the only caveat to that is the `-c` flag which is used by the "autostart" functionality to show the window when Autokey first starts.

**Reply by Elliria** [reply]
Okay, no worries on either of those. @josephj11 pointed out that we can deprecate **-l** rather than removing it and note that it will be removed in the future, and we haven't raised an eyebrow at **-c** and shouldn't ever need to.

**Comment by Elliria** [comment]
Once we're sure how we want to do it, I'd be happy to make those changes on the **develop** branch. That would be my first commit to change the actual behavior of AutoKey.

**Comment by Elliria** [comment]
I did a bit of research and although it would be a bit odd to have two short command-line switches and one long one together, it's not forbidden, so that means we can deprecate **-l** while leaving it in place. I'll create an issue for this and then make the change for the next version of AutoKey.

**Comment by ineuw** [comment]
Linux Mint Cinnamon also works by the same rules.
```
~$ autokey-gtk -V 
or 
~$ autokey-gtk --version
generate 
autokey-gtk Version 0.96.0
```

**Comment by josephj11** [comment]
@ineuw That code is completely internal to AutoKey, so it's independent of any distro.
